honey-token-tools/disseminate_honeytokens.py

```diff
@@ -0,0 +1,224 @@
+        )
+
+    try:
+        vcs_client = get_client_for_vcs(args.vcs, vcs_url, vcs_token)
+        vcs_client.validate_credentials()
+
+        # validate access to all the repositories
+        vcs_repos = [vcs_client.get_repository_info(repo) for repo in repos]
+
+        gg_client = GGClient(gitguardian_url, gitguardian_token)
+        gg_client.validate_credentials()
+
+        disseminate_honeytokens(vcs_client, gg_client, vcs_repos, args.output)
+
+    except (CredentialsValidationError, RepositoryAccessError) as error:
+        parser.error(error.message)
+
+
+def disseminate_honeytokens(
+    vcs_client, gg_client: GGClient, repos: list[RepositoryInfo], output: str
+):
+    result_output = dict()
+    for repo in repos:
+        # 1. create honey token with context
+        try:
+            data = gg_client.create_honey_token_with_context(repo)
+        except HoneyTokenCreationError as error:
+            result_output[repo.name] = {
+                "url": None,
+                "ok": False,
+                "error": f"Failed to create honey token: {error.message}",
+            }
+            continue
+        except Exception as error:
+            result_output[repo.name] = {
+                "url": "",
+                "ok": False,
+                "error": "Failed to create honey token: unknown error " + str(error),
+            }
+            continue
+
+        # 2. create pull request
+        pr_info = PullRequestInfo(
+            content=data["content"],
+            commit_message=data["suggested_commit_message"],
+            branch=get_branch_name_from_commit_message(
+                data["suggested_commit_message"]
+            ),
+            filename=data["filename"],
+        )
+
+        honey_token_id = data["honeytoken_id"]
+
+        try:
+            pull_request_url = vcs_client.create_pull_request(repo, pr_info)
+        except PullRequestCreationError as error:
+            # revoke honey token if pull request creation failed
+            ht_revoked = gg_client.revoke_honey_token(honey_token_id)
+            result_output[repo.name] = {
+                "url": None,
+                "ok": False,
+                "error": f"Failed to create pull request. {error.message}. Honey token {honey_token_id} was created "
+                + ("but revoked." if ht_revoked else "and not revoked."),
+            }
+            continue
+        except Exception as error:
+            # revoke honey token if pull request creation failed
+            ht_revoked = gg_client.revoke_honey_token(honey_token_id)
+            result_output[repo.name] = {
+                "url": None,
+                "ok": False,
+                "error": f"Failed to create pull request: unknown error {error}. Honey token {honey_token_id} was "
+                "created " + ("but revoked." if ht_revoked else "and not revoked."),
+            }
+            continue
+
+        result_output[repo.name] = {
```

Comment:
Should we return a honeytoken ID too ? This could be useful for further use with the GG API ?